skimage/feature/_hoghistogram.pyx

```diff
@@ -0,0 +1,135 @@
+                    float orientation_start, float orientation_end,
+                    int cell_columns, int cell_rows,
+                    int column_index, int row_index,
+                    int size_columns, int size_rows):
+    """Calculation of the cell's HOG value
+
+    Parameters
+    ----------
+    magnitude : ndarray
+        The gradient magnitudes of the pixels.
+    orientation : ndarray
+        Lookup table for orientations.
+    orientation_start : float
+        Orientation range start.
+    orientation_end : float
+        Orientation range end.
+    cell_columns : int
+        Pixels per cell (x).
+    cell_rows : int
+        Pixels per cell (y).
+    column_index : int
+        Block column index.
+    row_index : int
+        Block row index.
+    size_columns : int
+        Number of columns.
+    size_rows : int
+        Number of rows.
+
+    Returns
+    -------
+    total : float
+        The total HOG value.
+    """
+    cdef int cell_column, cell_row
+
+    cdef float total = 0.
+    for cell_row in range(-cell_rows/2, cell_rows/2):
+        cell_row_index = row_index + cell_row
+        if (cell_row_index < 0 or cell_row_index >= size_rows):
+            continue
+
+        for cell_column in range(-cell_columns/2, cell_columns/2):
+            cell_column_index = column_index + cell_column
+            if (cell_column_index < 0 or cell_column_index >= size_columns
+                or orientation[cell_row_index, cell_column_index]
+                    >= orientation_start
+                or orientation[cell_row_index, cell_column_index]
+                    < orientation_end): continue
+
+            total += magnitude[cell_row_index, cell_column_index]
+
+    return total
+
+def hog_histograms(cnp.float64_t[:, :] gradient_columns,
+                   cnp.float64_t[:, :] gradient_rows,
+                   int cell_columns, int cell_rows,
+                   int size_columns, int size_rows,
+                   int number_of_cells_columns, int number_of_cells_rows,
+                   int number_of_orientations,
+                   cnp.float64_t[:, :, :] orientation_histogram):
+    """Extract Histogram of Oriented Gradients (HOG) for a given image.
+
+    Parameters
+    ----------
+    gradient_columns : ndarray
+        First order image gradients (x).
+    gradient_rows : ndarray
+        First order image gradients (y).
+    cell_columns : int
+        Pixels per cell (x).
+    cell_rows : int
+        Pixels per cell (y).
+    size_columns : int
+        Number of columns.
+    size_rows : int
+        Number of rows.
+    number_of_cells_columns : int
+        Number of cells (x).
+    number_of_cells_rows : int
+        Number of cells (y).
+    number_of_orientations : int
+        Number of orientation bins.
+    orientation_histogram : ndarray
+        The histogram array which is modified in place.
+    """
+
+    cdef cnp.float64_t[:, :] magnitude = np.hypot(gradient_columns,
```

Comment:
Consider using typed memoryviews internally: `double[:, ::1]` for both `magnitude` and `orientation`. Must also change their input definitions in `cell_hog`. Ideally we'd do this almost globally, but that's a start.